src/poliastro/core/events.py

```diff
@@ -0,0 +1,53 @@
+import numpy as np
+from numba import njit as jit
+from numpy.linalg import norm
+
+from poliastro.core.elements import coe_rotation_matrix, rv2coe
+
+
+@jit
+def eclipse_function(k, u_, r_sec, R_sec, R_primary, umbra=True):
+    """Calculates a continuous shadow function.
+
+    Parameters
+    ----------
+    k: float
+        Standard gravitational parameter (km^3 / s^2).
+    u_: ~np.array
+        Satellite position and velocity vector with respect to the primary body.
+    r_sec: ~np.array
+        Position vector of the secondary body with respect to the primary body.
+    R_sec: float
+        Radius of the secondary body.
+    R_primary: float
+        Radius of the primary body.
+    umbra: bool
+        Whether to calculate the shadow function for umbra or penumbra, defaults to True
+        i.e. calculates for umbra.
+
+    """
+    # Plus or minus condition
+    pm = 1 if umbra else -1
+    p, ecc, inc, raan, argp, nu = rv2coe(k, u_[:3], u_[3:])
+
+    PQW = coe_rotation_matrix(inc, raan, argp)
+    P_, Q_ = PQW[:, 0], PQW[:, 1]
+
+    r_sec_norm = norm(r_sec)
+    beta = np.dot(P_, r_sec) / r_sec_norm
+    zeta = np.dot(Q_, r_sec) / r_sec_norm
+
+    delta_shadow = np.arcsin((R_sec - pm * R_primary) / r_sec_norm)
```

Comment:
I see you are applying the `arcsin` function and a few lines later the `sin`. Let use the same idea as in `cos_psi` variable, therefore:

```python
sin_delta_shadow = np.sin((R_sec - pm * R_primary) / r_sec_norm)
```

so only one function is being called instead of making round trip conversion.

Reply:
Thanks!